Tribes/src/app/_components/add-new-badge/add-new-badge.component.ts

```diff
@@ -13,25 +15,36 @@ export class AddNewBadgeComponent implements OnInit {
 
   goldAmount: number;
   costOfNewBuilding = COST_NEW_BUILDING;
+  soldier: Soldier;
   @Input() type: string;
   @Output() errorMessageEvent = new EventEmitter<string>();
+  costOfNewSoldier = COST_NEW_SOLDIER;
+  numberOfBarracks: number;
 
-  constructor(private resourceService: ResourceService, private buildingService: BuildingService) {
-    this.buildingService.beginConstruction.subscribe({
-      next: () => {
+  constructor(private resourceService: ResourceService, private buildingService: BuildingService, 
+    private soldiersService: SoldiersService) {
+      this.buildingService.beginConstruction.subscribe({
+        next: () => {
           this.getGoldAmount();
-      }
-    });
-   }
+        }
+      });
+      this.soldiersService.beginTraining.subscribe({
+        next: () => {
+          this.getGoldAmount();
+          this.getNumberOfBarracks();
+        }
+      });
```

Comment:
This also could use a.- some comments b.- the magical recalculation that Viola used.